samples/max-commit-delay.js

```diff
@@ -0,0 +1,80 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// sample-metadata:
+//  title: Executes request with max commit delay
+//  usage: node max-commit-delay.js <INSTANCE_ID> <DATABASE_ID> <PROJECT_ID>
+
+'use strict';
+
+function main(
+  instanceId = 'my-instance',
+  databaseId = 'my-database',
+  projectId = 'my-project-id'
+) {
+  // [START spanner_set_max_commit_delay]
+  // Imports the Google Cloud client library.
+  const {Spanner, protos} = require('@google-cloud/spanner');
+
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // const projectId = 'my-project-id';
+  // const instanceId = 'my-instance';
+  // const databaseId = 'my-database';
+
+  // Creates a client.
+  const spanner = new Spanner({
+    projectId: projectId,
+  });
+
+  async function spannerSetMaxCommitDelay() {
+    // Gets a reference to a Cloud Spanner instance and database.
+    const instance = spanner.instance(instanceId);
+    const database = instance.database(databaseId);
+
+    // Instantiate Spanner table objects.
+    const albumsTable = database.table('Albums');
+
+    // Updates rows in the Venues table.
+    try {
+      const [response] = await albumsTable.upsert(
```

Comment:
IIRC we were planning to use transactional example here . If that is true then you can try using `database.runTransaction` . Refer https://cloud.google.com/spanner/docs/samples/spanner-read-write-transaction#spanner_read_write_transaction-nodejs

Reply:
I believe I have fixed this. I'm not super familiar with NodeJS, so please double check and let me know if I have set this up incorrectly.

Thanks!